### PR TITLE
Add HTTP attributes support for AWS lambda function wrapper

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Related to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1773

This PR implements HTTP server spec for lambdas that use `APIGatewayProxyRequestEvent` as input.

### CODE REVIEW

File: instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/HttpSpanAttributes.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+final class HttpSpanAttributes {
```
Comment (anuraaga) [inline instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/HttpSpanAttributes.java:24]: Do you think it makes sense to have a `AwsLambdaGatewayProxyTracer extends HttpServerTracer`?

Comment (mateuszrzeszutek) [reply instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/HttpSpanAttributes.java:24]: I'd considered choosing different tracers based on the input class, but `AwsLambdaTracer` has some AWS specific logic (context propagation, `faas.*` attributes) that I'd need to duplicate in a new tracer (which would extend `HttpServerTracer`); I'd have to put the AWS logic in a separate helper class if I were to do that.
So out of the two choices: AWS logic in tracer vs HTTP logic in tracer I'd decided to put the HTTP things in the second place; it seemed to me that it'd be easier to add support for other event types this way (share the same tracer class with all the AWS things, separate strategy for additional attributes)

**Comment by iNikem** [comment]
@anuraaga nobody review this but you :) Or should I just merge it?

**Review by anuraaga — APPROVED** [review]
Oops sorry for missing - I'm a bit worried if the maintenance or http conventions is harder with this approach, e.g., when adding new http sem conventions. But I think it's fine for now anyways

**Pull request merged**